src/spells.cpp

```diff
@@ -720,7 +720,7 @@ bool Spell::playerRuneSpellCheck(Player* player, const Position& toPos)
 	}
 
 	if (aggressive && needTarget && topVisibleCreature && player->hasSecureMode()) {
-		const Player* targetPlayer = topVisibleCreature->getPlayer();
+		std::shared_ptr<const Player> targetPlayer = topVisibleCreature->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
		const auto targetPlayer = topVisibleCreature->getPlayer();
```